libgnucash/backend/dbi/gnc-backend-dbi.cpp

```diff
@@ -599,28 +616,29 @@ adjust_sql_options (dbi_conn connection)
 template <DbType Type> bool
 drop_database(dbi_conn conn, const UriStrings& uri)
 {
+    // The user may have CREATE privileges even if it does not have access to the "root_db"
     const char *root_db;
```

Comment:
root_db is used only in the DBI_PGSQL block so its decl might as well go there too.